module/data/actor/templates/common.mjs

```diff
@@ -130,13 +130,14 @@ export default class CommonTemplate extends ActorDataModel.mixin(CurrencyTemplat
   prepareAbilities({ rollData={}, originalSaves }={}) {
     const flags = this.parent.flags.dnd5e ?? {};
     const { prof = 0, ac } = this.attributes ?? {};
+    for ( const abl of Object.values(this.abilities) ) {
+      if ( flags.diamondSoul ) abl.proficient = 1;  // Diamond Soul is proficient in all saves
+      abl.mod = Math.floor((abl.value - 10) / 2);
+    }
```

Comment:
Since `diamondSoul` isn't actually necessary for this, we can just leave it down in the main loop and simplify this to a single line:

```suggestion
    Object.values(this.abilities).forEach(a => a.mod = Math.floor((a.value - 10) / 2));
```